src/common/net/fetch-client.ts

```diff
@@ -151,13 +200,74 @@ export class FetchHttpClient extends HttpClient implements HttpClientInterface {
 
     return new FetchHttpClientResponse(res);
   }
+
+  private async fetchRequestWithRetry(
+    url: string,
+    method: string,
+    body?: any,
+    headers?: RequestHeaders,
+  ): Promise<HttpClientResponseInterface> {
+    let response: HttpClientResponseInterface;
+    let requestError: any = null;
+    let retryAttempts = 1;
+
+    const makeRequest = async (): Promise<HttpClientResponseInterface> => {
+      try {
+        response = await this.fetchRequest(url, method, body, headers);
```

Comment:
Nice! Now we only need to maintain the one version of `fetchRequest`. Can we do the same thing for the node client as well?

Reply:
Discussed via Slack that doing this for the node client would require a bigger refactor of the `nodeRequest` method (since the node client uses callbacks instead of async/await). To de-risk this PR (already quite large/risky), we're not going to do this.